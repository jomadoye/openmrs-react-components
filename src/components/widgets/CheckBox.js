import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'react-bootstrap';

class CheckBox extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleToggle = this.handleToggle.bind(this);
  }

  componentDidMount() {
    const { input } = this.props;
    if (typeof input !== 'undefined') {
      const { onChange } = input;
      onChange('');
    }
  }

  handleToggle(e) {
    const { input } = this.props;
    if (typeof input !== 'undefined') {
      if (e.target.checked) {
        this.props.input.onChange(e.target.value);
      } else {
        this.props.input.onChange('');
      }
    }
  }

  render() {
    const { title, input, checkBoxValue } = this.props;
    return (
      <Checkbox
        {...input}
        onChange={this.handleToggle}
        value={!!(checkBoxValue.uuid) ? checkBoxValue.uuid : checkBoxValue}
      >{title}</Checkbox>
    );
  }
}

CheckBox.defaultProps = {
  title: "",
  checkBoxValue: "",
};

CheckBox.propTypes = {
  checkBoxValue: PropTypes.string.isRequired,
  input: PropTypes.object,
  title: PropTypes.string.isRequired,
};

export default CheckBox;
