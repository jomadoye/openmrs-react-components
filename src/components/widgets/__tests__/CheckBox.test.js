import React from 'react';
import renderer from 'react-test-renderer';
import CheckBox from '../CheckBox';

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = renderer.create(<CheckBox {...props} />);
  }
  return mountedComponent;
};

describe('Component: CheckBox', () => {
  beforeEach(() => {
    props = {
    };
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    const component = getComponent();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
