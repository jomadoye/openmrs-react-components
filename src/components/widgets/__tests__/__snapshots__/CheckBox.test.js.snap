// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CheckBox renders properly 1`] = `
<div>
  <div
    className="checkbox"
    style={undefined}
  >
    <label
      title=""
    >
      <input
        checked={false}
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      
    </label>
  </div>
</div>
`;
